tee-worker/identity/litentry/core/omni-account/src/repository.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{AccountId, Error, Header, OmniAccountMembers, OmniAccounts, ParentchainId};
+use alloc::vec::Vec;
+use frame_support::storage::storage_prefix;
+use itp_ocall_api::EnclaveOnChainOCallApi;
+use itp_storage::{
+	decode_storage_key, extract_blake2_128concat_key, storage_map_key, StorageHasher,
+};
+use litentry_primitives::{GetAccountStoreHash, MemberAccount};
+
+pub trait GetAccountStoresRepository {
+	fn get_by_owner(&self, owner: AccountId) -> Result<OmniAccountMembers, Error>;
+	fn get(&self) -> Result<OmniAccounts, Error>;
+}
+
+pub struct OmniAccountRepository<OCallApi: EnclaveOnChainOCallApi> {
+	ocall_api: OCallApi,
+	header: Header,
+}
+
+impl<OCallApi: EnclaveOnChainOCallApi> OmniAccountRepository<OCallApi> {
+	pub fn new(ocall_api: OCallApi, header: Header) -> Self {
+		Self { ocall_api, header }
+	}
+
+	pub fn set_header(&mut self, header: Header) {
+		self.header = header;
+	}
+}
+
+impl<OCallApi: EnclaveOnChainOCallApi> GetAccountStoresRepository
+	for OmniAccountRepository<OCallApi>
+{
+	fn get_by_owner(&self, owner: AccountId) -> Result<OmniAccountMembers, Error> {
+		let storage_key = storage_map_key(
+			"OmniAccount",
+			"AccountStore",
+			&owner,
+			&StorageHasher::Blake2_128Concat,
+		);
+		let storage_entry = self
+			.ocall_api
+			.get_storage_verified(storage_key, &self.header, &ParentchainId::Litentry)
+			.map_err(|_| Error::OCallApiError("Failed to get storage"))?;
+		let member_accounts: Vec<MemberAccount> =
+			storage_entry.value().to_owned().ok_or(Error::NotFound)?;
+		let account_store_hash = member_accounts.hash();
+
+		Ok(OmniAccountMembers { member_accounts, hash: account_store_hash })
+	}
+
+	fn get(&self) -> Result<OmniAccounts, Error> {
+		let account_store_key_prefix = storage_prefix(b"OmniAccount", b"AccountStore");
+		let account_store_storage_keys_response = self
+			.ocall_api
+			.get_storage_keys(account_store_key_prefix.into())
```

Comment:
This should get keys at `&self.header` or it might be querying different state here and in following `get_multiple_storages_verified`.